ci/infra/testrunner/tests/test_deployments.py

```diff
@@ -24,3 +21,13 @@ def test_nginx_deployment(deployment, kubectl):
     r = requests.get(url)
 
     assert "Welcome to nginx" in r.text
+
+    # Cleanup
+    kubectl.run_kubectl("delete --wait --timeout=60s service/nginx")
+    kubectl.run_kubectl("delete --wait --timeout=60s deployments/nginx")
+
+    with pytest.raises(Exception):
+        kubectl.run_kubectl("get service/nginx")
+
+    with pytest.raises(Exception):
+        kubectl.run_kubectl("get deployments/nginx")
```

Comment:
I would prefer to check directly if the service/deployment is present and not rely on exceptions (which may have a different cause) 

```
services = kubectl get services  -o jsonpath=' { .items[*].metadata.name }'.`
assert 'nginx not in services`
```